src/objects/Mesh.js

```diff
@@ -330,8 +334,6 @@ function checkIntersection( object, material, raycaster, ray, pA, pB, pC, point
 
 	const distance = raycaster.ray.origin.distanceTo( _intersectionPointWorld );
 
-	if ( distance < raycaster.near || distance > raycaster.far ) return null;
```

Comment:
Maybe we should return this line because it is possible that some intersections pass the first check (bounding sphere) but fail this one.

Reply:
Yes this check will still be needed. The closest surface of the sphere could be in the range `[near, far]` while a triangle in the sphere is outside of that range.

Reply:
correct!